proof/invariant-abstract/DetSchedSchedule_AI.thy

```diff
@@ -26446,10 +26458,12 @@ lemma handle_event_preemption_path_valid_sched:
   done
 
 lemma call_kernel_valid_sched:
-  "\<lbrace>\<lambda>s. valid_sched s \<and> invs s \<and> valid_machine_time s \<and> schact_is_rct s \<and> ct_not_in_release_q s
-        \<and> ct_released s \<and> (ct_running s \<or> ct_idle s) \<and> consumed_time_bounded s
-        \<and> cur_sc_active s \<and> cur_sc_offset_ready (consumed_time s) s \<and> current_time_bounded 5 s
-        \<and> cur_sc_offset_sufficient (consumed_time s) s \<and> (e \<noteq> Interrupt \<longrightarrow> ct_running s)\<rbrace>
+  "\<lbrace>\<lambda>s. valid_sched s \<and> invs s \<and> schact_is_rct s
+        \<and> cur_sc_active s \<and> ct_not_in_release_q s
+        \<and> (ct_running s \<or> ct_idle s) \<and> (e \<noteq> Interrupt \<longrightarrow> ct_running s)
+        \<and> valid_machine_time s \<and> current_time_bounded 5 s \<and> consumed_time_bounded s
```

Comment:
`current_time_bounded  5` looks like an oddly specific number, I'm guessing some WCET multiple from the end of time, but e should at least have a named constant for it.

Reply:
Yeah `current_time_bounded n` roughly says we are the WCET plus `n` times the `MAX_PERIOD` from the end of time.

We have wondered before whether all instances of `current_time_bounded n` can be replaced with `current_time_bounded 5`. Some of the proofs for handling overrun are very tricky, so it's hard to say without trying it. Would that be the best outcome? Would you be happy for me to merge these PRs for these invariants, and to come back to this? At the very least, yes, we can make a named constant for this number

Reply:
Happy for you to clean it up later, yes. As long as it actually happens :-)